test/app/fiber_channel.result

```diff
@@ -594,7 +594,7 @@ collectgarbage('collect')
 ---
 - 0
 ...
-refs -- must be zero
+refs -- may be non-zero due to compat.fiber_channel_close_mode (graceful close)
 ---
-- 0
+- 8
```

Comment:
The test case lost his sense this way, I think. How about testing the new condition, when the channel decreases refcounters of its objects (at GC of the channel object itself)?

Something like this:

```lua
refs == 8 or refs
ch = nil
collectgarbage()
refs == 0 or refs
```

And, of course, reflect current state and link newer issues in the comment for the test case (above).

Reply:
Done. Note that we need a double gc call due to [resurrection](https://stackoverflow.com/questions/28320213/why-do-we-need-to-call-luas-collectgarbage-twice).